packages/run-protocol/src/vaultFactory/vault.js

```diff
@@ -36,7 +36,22 @@ export const VaultState = {
   CLOSED: 'closed',
 };
 
-/** @type {MakeVaultKit} */
+/**
+ * @param {ContractFacet} zcf
+ * @param {{reallocateReward: ReallocateReward;
+    getCollateralBrand: () => Brand;
+    getLiquidationMargin(): Ratio;
+    getInitialMargin(): Ratio;
+    getLoanFee(): Ratio;
+    getInterestRate(): Ratio;
+    getChargingPeriod(): bigint;
+    getRecordingPeriod(): bigint;
+    getCollateralQuote(): Promise<PriceQuote>}} manager
```

Comment:
this is an example of something where I'd rather see a separate VaultManager type. This overall change still seems like that should be possible.

Reply:
Happy to engage on this. We could restore `InnerVaultManager` type. First I'd like to understand the motivation. Is it because that's a thing we should name or just that this is hard to read?

If it's about naming the thing, I contend shouldn't be a named type because it shouldn't be a concept. Here it's just a bunch of stuff that's being passed around. I especially think we shouldn't have the `GetVaultParams` and union because that merging is an implementation detail.

If it's just for readability, we could put a type in this file that doesn't get exported. However I'd still offer that the reason this is hard to read may be mostly due to how it's factored. Having it like this highlights it as something to address.

Reply:
I'm also not a big fan of complex inline param types, maybe with the exception of option bags / powers.

Seeing how this method has 5 different params, this manager doesn't seem to fit that definition. If it's indeed behaving like an aggregation, I'd like to understand why it's aggregated separately from the other params.